### ISSUE TITLE
[v4 α2] Regression on theme styleOverrides

### ISSUE THREAD

**Issue by smeng9** [marmelab/react-admin#7164]
After fixing dependency issue in latest alpha demo, there seems to be some regression on theme overrides.

**What you were expecting:**

The Posters Galore Application
For example, in v3 the MenuItemLink should have a left border when active
![reactadminbug](https://user-images.githubusercontent.com/38666763/151688029-e4c25e69-ee73-427e-aca5-51136540a239.PNG)

**What happened instead:**

In v4 there is no left border even when we manually override it.
Also AppBar background color is also not effective.
![react-adminnobug](https://user-images.githubusercontent.com/38666763/151688050-d8ed41fb-d966-451b-bdc9-49ca71906f6b.PNG)

**Steps to reproduce:**

Clone the https://github.com/marmelab/react-admin/tree/next/examples/demo and resolve all dependency issue in codesandbox

**Related code:**

```
https://codesandbox.io/s/exciting-bhabha-82u9j?file=/src/layout/themes.ts:1928-1950
```

**Other information:**

**Environment**

* React-admin version: 4.0.0-alpha.2
* Last version that did not exhibit the issue (if applicable): 3.19.7
* React version: 17.0.2
* Browser: Edge 97
* Stack trace (in case of a JS error): N/A

**Comment by emulienfou** [comment]
Same issue since **4.0.0-alpha.1**: https://github.com/marmelab/react-admin/issues/7079

**Comment by WiXSL** [comment]
Fixed by #7166
